linkerd/tls/src/server.rs

```diff
@@ -107,27 +84,27 @@ impl<L, P, N> NewDetectTls<L, P, N> {
     }
 }
 
-impl<T, L, P, N> NewService<T> for NewDetectTls<L, P, N>
+impl<T, L, P, N> NewService<(T, Option<ServerName>)> for NewDetectTls<L, P, N>
```

Comment:
Using a tuple-target is a bit of an anti-pattern: it's fragile and doesn't really permit any intermediate layers to work. It may be appropriate here if we can bundle it up so that it's opaque to the caller, but it's preferable to use a Param or ExtractParam to get a ServerName from the target.